File: app/lib/api/personal_events_api.dart
import '../models/calendar_event.dart';
import 'api_client.dart';

class PersonalEventsApi {
  PersonalEventsApi(this._client);
  final ApiClient _client;

  /// `GET /api/me/events?from=&to=` — private events for the signed-in user.
  Future<List<CalendarEvent>> fetchPersonalEvents({
    required String from,
    required String to,
  }) async {
    final data = await _client
        .requestJson('/api/me/events', query: {'from': from, 'to': to});
    final raw = data['events'];
    if (raw is List) {
      return raw
          .whereType<Map>()
          .map((e) => CalendarEvent.fromJson(Map<String, dynamic>.from(e)))
          .toList();
    }
    return const [];
  }

  Future<CalendarEvent> createPersonalEvent(Map<String, dynamic> payload) async {
    final data = await _client.requestJson('/api/me/events',
        method: 'POST', body: payload);
    return CalendarEvent.fromJson(Map<String, dynamic>.from(data['event']));
  }

  Future<CalendarEvent> patchPersonalEvent(
      String id, Map<String, dynamic> payload) async {
    final data = await _client.requestJson('/api/me/events/$id',
        method: 'PATCH', body: payload);
    return CalendarEvent.fromJson(Map<String, dynamic>.from(data['event']));
  }

  Future<void> removePersonalEvent(String id) async {
    await _client.requestJson('/api/me/events/$id', method: 'DELETE');
  }
}

File: src/utils/personalEventsApi.js
import { apiFetch } from '../auth/AuthContext';

function apiError(res, data) {
    const err = new Error((data && data.error) || `HTTP ${res.status}`);
    err.status = res.status;
    err.code = data && data.error;
    return err;
}

/** Fetch the logged-in user's personal events for a date range. */
export async function fetchPersonalEvents({ from, to }) {
    const params = new URLSearchParams({ from, to });
    const res = await apiFetch(`/api/me/events?${params}`);
    const data = await res.json().catch(() => ({}));
    if (!res.ok) throw apiError(res, data);
    return Array.isArray(data.events) ? data.events : [];
}

export async function createPersonalEvent(partial) {
    const res = await apiFetch('/api/me/events', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify(partial),
    });
    const data = await res.json().catch(() => ({}));
    if (!res.ok) throw apiError(res, data);
    return data.event;
}

export async function patchPersonalEvent(id, partial) {
    const res = await apiFetch(`/api/me/events/${encodeURIComponent(id)}`, {
        method: 'PATCH',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify(partial),
    });
    const data = await res.json().catch(() => ({}));
    if (!res.ok) throw apiError(res, data);
    return data.event;
}

export async function removePersonalEvent(id) {
    const res = await apiFetch(`/api/me/events/${encodeURIComponent(id)}`, {
        method: 'DELETE',
    });
    const data = await res.json().catch(() => ({}));
    if (!res.ok) throw apiError(res, data);
    return data.id;
}
